### PR TITLE
fix(Link): [FX-2231] fix on hover style

### PR DESCRIPTION
**Pull request by vshyrokov**
[FX-2231](https://toptal-core.atlassian.net/browse/FX-2231)

### Description

Updated on hover style when textDecoration is none.

### How to test

- when hovering over the link with textDecoration='none', the underline shouldn't appear.
- when hovering over the disabled link with textDecoration='none', the underline shouldn't appear.
- when hovering over the disabled link with textDecoration='underline', the underline shouldn't disappear

[link](https://picasso.toptal.net/FX-2231-fix-on-hover-style-in-link/?path=/story/components-link--link#text-decoration)

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- `N/A` Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Comment by vshyrokov** [comment]
@toptal-bot run visual

**Comment by Rootre** [comment]
Looks good to me👍

File: packages/picasso/src/Link/story/Disabled.example.tsx [hunk 1]
```diff
@@ -1,16 +1,29 @@
...
+        onClick={() => {
+          console.log('foobar')
+        }}
+        disabled
+      >
+        Underline Link
+      </Link>
+    </Container>
+    <Container inline right='large'>
+      <Link
```
Comment (TomasSlama) [inline packages/picasso/src/Link/story/Disabled.example.tsx:17]: As Milos said in base channel, Link with textDecoration none should not be used standalone. So I don't think this example is correct. `textDecoration='none'`  should be used only when Link is wrapping another component so we don't change text styling inside that component.

**Review by TomasSlama — COMMENTED** [review]
The code it self looks good, but I would change the storybook example.

File: packages/picasso/src/Link/story/Disabled.example.tsx (hunk 1 above)
Comment (vshyrokov) [reply packages/picasso/src/Link/story/Disabled.example.tsx:17]: Hey @TomasSlama, the biggest problem for textDecoration='none', is when we hovered over the link, appeared underline, it's not a correct behaviour

Comment (vshyrokov) [reply packages/picasso/src/Link/story/Disabled.example.tsx]: updated the code

File: packages/picasso/src/Link/test.tsx [hunk 2]
```diff
@@ -98,4 +154,38 @@ describe('Link', () => {
 
     expect(getByTestId('foo')).not.toHaveAttribute('href')
   })
+
+  it('does not have underline when textDecoration is none', () => {
```
Comment (dmaklygin1) [inline packages/picasso/src/Link/test.tsx:158]: we should wrap each case with `describe` block. https://toptal-core.atlassian.net/wiki/spaces/ENG/pages/1925971969/Frontend+Testing+-+Testing+of+React+Components#Testing-plan

File: packages/picasso/src/Link/__snapshots__/test.tsx.snap [hunk 3]
```diff
@@ -1,5 +1,53 @@
...
+</div>
+`;
+
+exports[`Link does not have underline when textDecoration is underline on hover 1`] = `
+<div>
+  <div
+    class="Picasso-root"
+  >
+    <a
+      class="MuiTypography-root MuiLink-root MuiLink-underlineHover PicassoLink-root PicassoLink-fontSizeInitial PicassoLink-textDecorationUnderline MuiTypography-colorPrimary"
```
Comment (dmaklygin1) [inline packages/picasso/src/Link/__snapshots__/test.tsx.snap:41]: Actually, there are two classes are related to underline: MuiLink-underlineHover, PicassoLink-textDecorationUnderline
However, this link should not show underline

File: packages/picasso/src/Link/test.tsx [hunk 4]
```diff
@@ -57,6 +57,62 @@ describe('Link', () => {
...
+        download='filename'
+        href='https://toptal.com/filename.txt'
+        textDecoration='none'
+        disabled
+      >
+        Please verify your email
+      </Link>
+    )
+
+    expect(container).toMatchSnapshot()
```
Comment (dmaklygin1) [inline packages/picasso/src/Link/test.tsx:74]: I would avoid using snapshots, btw :) we have rules

File: packages/picasso/src/Link/test.tsx [hunk 5]
```diff
@@ -98,4 +154,38 @@ describe('Link', () => {
...
   })
+
+  it('does not have underline when textDecoration is none', () => {
+    const { container } = render(
+      <Link data-testid='foo' href='https://foo.bar' textDecoration='none'>
+        Test
+      </Link>
+    )
+
+    expect(container).toMatchSnapshot()
```
Comment (dmaklygin1) [inline packages/picasso/src/Link/test.tsx:165]: How would you make sure the final component does not have `underline`, it's not clear

**Comment by TomasSlama** [comment]
@vshyrokov Can you please also remove this example from the storybook? 👉 https://picasso.toptal.net/FX-2231-fix-on-hover-style-in-link/?path=/story/components-link--link#text-decoration as Milos suggested in our discussion?

**Review by teimurjan — APPROVED** [review]
👍

**Review by dmaklygin1 — APPROVED** [review]

**Review by TomasSlama — APPROVED** [review]
GJ 👍

**Pull request merged**